### DISCUSSION TITLE
Preview upcoming days/weeks

### DISCUSSION THREAD

**Discussion by sphuo5g** [iSoron/uhabits#2015] [Feature Requests]
As a user with multiple long-term habits, I need to be able to check the date a habit is next due. This if especially required when a task takes some preparation, i.e. a task for which it’s sometimes necessary to organize some appointment first.

The date could for example be shown next to the indicator showing the interval, i.e. "every 90 days, next due 2024-07-03" or "every 90 days, next due tomorrow".

**Comment by KyleSCraig** [comment]
At the moment if you scroll down to the the streaks it does forecast out 30 days so for example if I tick a task every 90 days, it says the streak goes until 30 days from now. If it's due in 10 days, then it will say the streak goes until 9 days from now.
I do think that it would be good to have the ability to enter a date (not further out than 30 days anyway) and have it show if you do nothing until then what will be due. I believe that would be much easier to implement than what's described here and mean you don't have to enter the task to see. With manual sorting you can group all infrequent tasks together.
